### PR TITLE
[FIX] payment, payment_authorize: payment error flow

### PR DESCRIPTION
**Pull request by bouvyd**

### CODE REVIEW

**Comment by dbo-odoo** [comment]
@robodoo rebase-ff

File: addons/payment/controllers/portal.py [hunk 1]
```diff
@@ -331,6 +331,9 @@ def confirm(self, **kw):
             elif tx.state == 'pending':
                 status = 'warning'
                 message = tx.acquirer_id.pending_msg
+            else:
+                status = 'danger'
+                message = tx.state_message or _('An error occured during the processing of this payment')
```
Comment (dbo-odoo) [inline addons/payment/controllers/portal.py:336]: note: this string is already in the `.pot` file, no need to add it

File: addons/payment_authorize/models/authorize_request.py [hunk 2]
```diff
@@ -226,11 +226,15 @@ def auth_and_capture(self, token, amount, reference):
...
 
-        return {
+        result = {
             'x_response_code': response.get('transactionResponse', {}).get('responseCode'),
             'x_trans_id': response.get('transactionResponse', {}).get('transId'),
             'x_type': 'auth_capture'
         }
+        errors = response.get('transactionResponse', {}).get('errors')
+        if errors:
+            result['x_response_reason_text'] = '\n'.join([e.get('errorText') for e in errors])
```
Comment (AntoineVDV) [inline addons/payment_authorize/models/authorize_request.py:236]: Nitpicking
```suggestion
            result['x_response_reason_text'] = '\n'.join(e.get('errorText') for e in errors)
```
My comment was scrapped somehow so here it is.

**Comment by dbo-odoo** [comment]
@robodoo r+

**Pull request closed without merging**